File: app/src/androidTest/java/cgeo/geocaching/brouter/util/SortedHeapTest.java
package cgeo.geocaching.brouter.util;

import java.util.Random;

import org.junit.Assert;
import org.junit.Test;

public class SortedHeapTest {
    @Test
    public void sortedHeapTest1() {
        final SortedHeap<String> sh = new SortedHeap<>();
        final Random rnd = new Random();
        for (int i = 0; i < 100000; i++) {
            int val = rnd.nextInt(1000000);
            sh.add(val, "" + val);
            val = rnd.nextInt(1000000);
            sh.add(val, "" + val);
            sh.popLowestKeyValue();
        }

        int cnt = 0;
        int lastval = 0;
        for (; ; ) {
            final String s = sh.popLowestKeyValue();
            if (s == null) {
                break;
            }
            cnt++;
            final int val = Integer.parseInt(s);
            Assert.assertTrue("sorting test", val >= lastval);
            lastval = val;
        }
        Assert.assertTrue("total count test", cnt == 100000);

    }

    @Test
    public void sortedHeapTest2() {
        final SortedHeap<String> sh = new SortedHeap<>();
        for (int i = 0; i < 100000; i++) {
            sh.add(i, "" + i);
        }

        int cnt = 0;
        int expected = 0;
        for (; ; ) {
            final String s = sh.popLowestKeyValue();
            if (s == null) {
                break;
            }
            cnt++;
            final int val = Integer.parseInt(s);
            Assert.assertTrue("sequence test", val == expected);
            expected++;
        }
        Assert.assertTrue("total count test", cnt == 100000);

    }
}

File: main-kotlin/src/androidTest/kotlin/cgeo/geocaching/brouter/util/SortedHeapTest.kt
// Auto-converted from Java to Kotlin
// WARNING: This code requires manual review and likely has compilation errors
// Please review and fix:
// - Method signatures (parameter types, return types)
// - Field declarations without initialization
// - Static members (use companion object)
// - Try-catch-finally blocks
// - Generics syntax
// - Constructors
// - And more...

package cgeo.geocaching.brouter.util

import java.util.Random

import org.junit.Assert
import org.junit.Test

class SortedHeapTest {
    @Test
    public Unit sortedHeapTest1() {
        val sh: SortedHeap<String> = SortedHeap<>()
        val rnd: Random = Random()
        for (Int i = 0; i < 100000; i++) {
            Int val = rnd.nextInt(1000000)
            sh.add(val, "" + val)
            val = rnd.nextInt(1000000)
            sh.add(val, "" + val)
            sh.popLowestKeyValue()
        }

        Int cnt = 0
        Int lastval = 0
        for (; ; ) {
            val s: String = sh.popLowestKeyValue()
            if (s == null) {
                break
            }
            cnt++
            val val: Int = Integer.parseInt(s)
            Assert.assertTrue("sorting test", val >= lastval)
            lastval = val
        }
        Assert.assertEquals("total count test", 100000, cnt)

    }

    @Test
    public Unit sortedHeapTest2() {
        val sh: SortedHeap<String> = SortedHeap<>()
        for (Int i = 0; i < 100000; i++) {
            sh.add(i, "" + i)
        }

        Int cnt = 0
        Int expected = 0
        for (; ; ) {
            val s: String = sh.popLowestKeyValue()
            if (s == null) {
                break
            }
            cnt++
            val val: Int = Integer.parseInt(s)
            Assert.assertEquals("sequence test", val, expected)
            expected++
        }
        Assert.assertEquals("total count test", 100000, cnt)

    }
}
